crates/ruff/src/rules/flake8_simplify/rules/reimplemented_builtin.rs

```diff
@@ -230,6 +230,10 @@ pub(crate) fn convert_for_loop_to_any_all(
     stmt: &Stmt,
     sibling: Option<&Stmt>,
 ) {
+    // Don't flag if the loop is in an async context (`any` and `all` are not async).
+    if checker.semantic().in_async_context() {
+        return;
+    }
```

Comment:
Could we instead check if the statement body contains an `await`? We have a `contains_await` utility in `rules/pyupgrade/rules/unpacked_list_comprehension.rs` that we could move to `helpers.rs`.

E.g., sometimes, it's okay to run this in an async function:

```py
async def f():
    # OK
    for x in iterable:
        if check(x):
            return True
    return False
```

Reply:
Yeah, that makes sense. The PR is a bit nuclear.

Reply:
Oh wait, we also have `is_async_generator` which is significantly simpler in `rules/flake8_comprehensions/rules/unnecessary_comprehension_any_all.rs`. We should be able to reuse that snippet (but we can just redefine it here).